lib/src/layer/tile_layer/tile_display.dart

```diff
@@ -45,15 +45,12 @@ abstract class TileDisplay {
   T? map<T>({
     T? Function(InstantaneousTileDisplay instantaneous)? instantaneous,
     T? Function(FadeInTileDisplay fadeIn)? fadeIn,
-  }) {
-    switch (runtimeType) {
-      case InstantaneousTileDisplay:
-        return instantaneous?.call(this as InstantaneousTileDisplay);
-      case FadeInTileDisplay:
-        return fadeIn?.call(this as FadeInTileDisplay);
-    }
-    return null;
-  }
+  }) =>
+      switch (this) {
+        InstantaneousTileDisplay() =>
+          instantaneous?.call(this as InstantaneousTileDisplay),
+        FadeInTileDisplay() => fadeIn?.call(this as FadeInTileDisplay),
+      };
```

Comment:
Prefer this implementation as this will avoid the usage of the `as` keyword:
```dart
sealed class TileDisplay {
  // ...

  @optionalTypeArgs
  T? map<T extends Object?>({
    T? Function(InstantaneousTileDisplay)? instantaneous,
    T? Function(FadeInTileDisplay)? fadeIn,
  }) {
    final tileDisplay = this;
    return switch (tileDisplay) {
      InstantaneousTileDisplay() => instantaneous?.call(tileDisplay),
      FadeInTileDisplay() => fadeIn?.call(tileDisplay),
    };
  }
}
```

Reply:
(Was my change.)
Resolved in 2b1b2f7, except for the (imo unnecessary) type narrowing on `T` and `@optionalTypeArgs` annotation. Also renamed it to `when` to better reflect what it does.